lib/plugins/fs-routes/router-template.js

```diff
@@ -0,0 +1,127 @@
+    return routes
+      .map(
+        route => `
+    {
+      path: ${JSON.stringify(route.path)},
+      component: () => import(${JSON.stringify(route.component)}),
+      ${route.children ? `children: [${renderRoutes(route.children)}]` : ``}
+    }`
+      )
+      .join(',')
+  }
+
+  return `
+  import Router from 'vue-router'
+
+  export default function () {
+    return new Router({
+      mode: 'history',
+      routes: [
+        ${renderRoutes(routes)}
+      ]
+    })
+  }`
+}
+
+// index.vue -> /
+// about.vue -> /about
+// user.vue -> /user
+// user/index.vue -> /user, child ''
+// user/friends.vue -> /user, child 'friends'
+// catalog/index.vue -> /catalog
+// catalog/specials.vue -> /catalog/specials
+
+async function collectRoutes({ dir, componentPrefix, parentPath }) {
+  const records = []
+  const recordsLookup = {}
+
+  function addRecord(path, props) {
+    if (!recordsLookup[path]) {
+      const record = {
+        path: filePathToRoutePath(path)
+      }
+      records.push(record)
+      recordsLookup[path] = record
+    }
+    Object.assign(recordsLookup[path], props)
+  }
+
+  for (const name of await fs.readdir(dir)) {
+    const fullName = path.join(dir, name)
+    const stats = await fs.stat(fullName) // eslint-disable-line no-await-in-loop
+    if (stats.isDirectory()) {
+      addRecord(name, { dir: name })
+    } else if (stats.isFile()) {
+      const match = name.match(/(.*)\.(vue|jsx?|ts)$/i) // TODO: pull from webpack?
+      if (match) {
+        const baseName = match[1]
+        const path = baseName.toLowerCase() === 'index' ? '' : baseName
+        addRecord(path, { file: name })
+      }
+    }
+  }
+
+  const sortedRecords = sortBy(records, record => [
+    record.path.indexOf(':') >= 0, // Dynamic routes go last
+    record.path
+  ])
+
+  const routes = []
+
+  for (const record of sortedRecords) {
+    const fullRoutePath = parentPath
+      ? path.join(parentPath, record.path)
+      : record.path
+    if (record.file) {
+      const route = {
+        path: fullRoutePath,
+        component: path.join(componentPrefix, record.file)
+      }
+      if (record.dir) {
+        // eslint-disable-next-line no-await-in-loop
+        route.children = await collectRoutes({
+          dir: path.join(dir, record.dir),
+          componentPrefix: path.join(componentPrefix, record.dir),
+          parentPath: ''
+        })
+      }
+      routes.push(route)
+    } else if (record.dir) {
+      routes.push(
+        // eslint-disable-next-line no-await-in-loop
+        ...(await collectRoutes({
+          dir: path.join(dir, record.dir),
+          componentPrefix: path.join(componentPrefix, record.dir),
+          parentPath: fullRoutePath
+        }))
+      )
+    }
+  }
+
+  return routes
+}
+
+function filePathToRoutePath(path) {
+  // Nuxt-compatible syntax: _item_id.vue
+  // egoist suggested syntax: [item_id].vue
+  return path.replace(/^_/, ':').replace(/\[(.*?)\]/g, ':$1')
+}
```

Comment:
I prefer using `[username].vue` syntax and ignoring paths starting with an underscore `_`, when I was using Nuxt.js I always want to put some page-related components in `pages` but don't know how to exclude them.

Reply:
Done. `_*` and `.*` are excluded, `[username].vue` is the only supported syntax.